CONTRIBUTING.md

```diff
@@ -48,8 +29,137 @@ git checkout joes_feature_branch
+
+#### macOS
+
+##### [Install Visual Studio for Mac](https://visualstudio.microsoft.com/vs/mac/)
+
+* During setup, install
+  * .NET Core
+  * Android
+  * iOS
+  * MacOS
+* In Visual Studio for Mac, select **Tools** > **SDK Manager** and install Android SDK with API 29.
+
+The steps above should enable you to compile `Libs.sln`. You will need a developer certificate to compile `LibsMacOS.sln`.
+
+### Fast build
+
+MSAL.NET supports several target frameworks, but most of the time contributors are only interested in one or two. To get MSAL to build for all frameworks, contributors will need a [hefty Visual Studio installation as well as several SDKs](#prerequisites-to-build-msalnet).
+
+To work around this requirement, open [`Microsoft.Identity.Client.csproj`](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/master/src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj) and comment out the targets you are not interested in. Keeping the pure .NET targets and eliminating UWP and Xamarin results in a fast build as well as the ability to run all unit tests.
+
+Visual Studio may need to be restarted to ensure that updated target frameworks take effect.
+
+### Visual Studio for Mac
+
+MSAL is a multi-target library and at the time of writing, Visual Studio for Mac is not able to understand and layout this project correctly. The library can still be built from the command line on macOS.
+
+### Visual Studio
+
+1. Load `LibsAndSamples.sln` for a bigger solution with lots of apps that showcase and exercise MSAL. Load `LibsNoSamples.sln` for a small solution that has the library and the tests.
+2. Build in Visual Studio (if configured) or via the command line with `msbuild /t:restore` and `msbuild`. If using the command line, developers might need to use the [Visual Studio Developer Command Prompt](/visualstudio/ide/reference/command-prompt-powershell).
+
+>**Note**
+>If you run into strong name validation issues, please [log a bug](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues). Workaround is to disable strong name validation on your dev box by running the following command in the Visual Studio Developer Command Prompt with Administrator permissions:
+>
+>```bash
+>sn -Vr *
+>```
+
+>**Note**
+>You won't be able to run the integration or automation tests because they require access to a Microsoft Key Vault instance which is only accessible to the MSAL.NET engineering team. These tests will run as part of our automation pipelines in GitHub.
```

Comment:
```suggestion
>You won't be able to run the integration or automation tests because they require access to a Azure Key Vault instance which is only accessible to the MSAL.NET engineering team. These tests will run as part of our automation pipelines in GitHub.
```